### PR TITLE
Mark group as optional in LocalObjectReference

### LINKED ISSUE #3322: Group is not optional in LocalObjectReference
LocalObjectReference group field is not marked optional. So, it can't be left unspecified or set to empty string.

In BackendTLSPolicy, we try to ref to a Secret and CRD fails validation check as group is ""

Previous conv in Slack: https://kubernetes.slack.com/archives/CR0H13KGA/p1725096986532939

Screenshot 2024-08-31 at 2 35 20 AM

### PR DESCRIPTION
**Pull request by tamalsaha**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
Make group as optional in LocalObjectReference

**Which issue(s) this PR fixes**:

Fixes #3322

**Does this PR introduce a user-facing change?**:

```release-note
Mark group as optional in LocalObjectReference
```

### CODE REVIEW

File: apis/v1/object_reference_types.go [hunk 1]
```diff
@@ -27,6 +27,8 @@ package v1
 type LocalObjectReference struct {
 	// Group is the group of the referent. For example, "gateway.networking.k8s.io".
 	// When unspecified or empty string, core API group is inferred.
+	//
+	// +optional
```
Comment (arkodg) [inline apis/v1/object_reference_types.go:31]: also need to make Group a pointer and add a `omitempty` tag

Comment (tamalsaha) [reply apis/v1/object_reference_types.go:31]: Done!

**Comment by arkodg** [comment]
this may not be a breaking change for users, but may be a breaking change for implementations, I still +1 this change to improve user experience (lesser config to write)

**Pull request closed without merging**